libcore/src/IO/GeoFileParser.cpp

```diff
@@ -191,9 +191,40 @@ bool GeoFileParser::LoadGeometry(Building * building)
                     xVertex = xVertex->NextSiblingElement("vertex")) {
                     double x1 = xmltof(xVertex->Attribute("px"));
                     double y1 = xmltof(xVertex->Attribute("py"));
+                    Point p1{x1, y1};
+
                     double x2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("px"));
                     double y2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("py"));
-                    subroom->AddWall(Wall(Point(x1, y1), Point(x2, y2), wall_type));
+                    Point p2{x2, y2};
+
+                    Wall wall{p1, p2, wall_type};
+                    subroom->AddWall(wall);
+
+                    if(wall_type == "track") {
+                        int trackID = xmltoi(
+                            xPolyVertices->Attribute("track_id"), std::numeric_limits<int>::min());
+                        if(trackID < 0) {
+                            //TODO warning
+                            continue;
+                        }
```

Comment:
No, this is for extracting the track in the geometry (platform edges). Therefore `trackID` should be >=0 and the track will be ignored otherwise. Added a warning with this message.